### PR TITLE
Fix resampling failure of last particle

### PR DESCRIPTION
**Pull request by nhatao**
In the master branch, last particle is not removed even when its probability is small enough.
The algorithm described in Probabilistic Robotics is implemented.

### CODE REVIEW

**Comment by at-wat** [comment]
## [[#790](https://travis-ci.org/at-wat/mcl_3dl/builds/684531866)] PASSED on noetic

Tested on Alpine ROS

**Comment by nhatao** [comment]
@at-wat PTAL.

File: test/src/test_pf.cpp [hunk 1]
```diff
@@ -235,6 +223,40 @@ TEST(Pf, ResampleFirstParticle)
...
+        };
+    const std::vector<float> states =
+        {
+            0.0f, 1.0f, 2.0f, 3.0f, 4.0f
+        };
+    const std::vector<float> expected_resampled_states =
+        {
+            1.0f, 2.0f, 3.0f, 4.0f, 4.0f
+        };
+    testResample(probs, states, expected_resampled_states);
```
Comment (at-wat) [inline test/src/test_pf.cpp:242]: Please add scoped trace.
https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#adding-traces-to-assertions

Comment (nhatao) [reply test/src/test_pf.cpp:243]: Done

**Comment by at-wat** [comment]
## [[#791](https://travis-ci.org/at-wat/mcl_3dl/builds/684553496)] PASSED on noetic

Tested on Alpine ROS

**Pull request merged**